src/dotnet-benchmark-client/Startup.cs

```diff
@@ -0,0 +1,205 @@
+
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                var allJobs = _jobs.GetAll();
+                var job = allJobs.FirstOrDefault();
+                if (job != null)
+                {
+                    var jobLogText = $"[ID:{job.Id} Connections:{job.Connections} Threads:{job.Threads} Duration:{job.Duration} Pipeline:{job.PipelineDepth}]";
+
+                    if (job.State == ClientState.Waiting)
+                    {
+                        // TODO: Race condition if DELETE is called during this code
+
+                        Log($"Starting job {jobLogText}");
+                        job.State = ClientState.Starting;
+
+                        Debug.Assert(process == null);
+
+                        Log($"Running job {jobLogText}");
+                        job.State = ClientState.Running;
+
+                        process = StartProcess(job);
+                    }
+                    else if (job.State == ClientState.Deleting)
+                    {
+                        Log($"Deleting job {jobLogText}'");
+
+                        Debug.Assert(process != null);
+
+                        process.WaitForExit();
+                        process.Dispose();
+                        process = null;
+
+                        _jobs.Remove(job.Id);
+                    }
+                }
+                await Task.Delay(100);
+            }
+        }
+
+        private static Process StartProcess(ClientJob job)
+        {
+            var tcs = new TaskCompletionSource<bool>();
+
+            var command = $"wrk -c {job.Connections} -t {job.Threads} -d {job.Duration}";
+            if (job.PipelineDepth > 0)
+            {
+                command += $" -s scripts/pipeline.lua";
+            }
+            command += $" {job.ServerBenchmarkUri}";
+            if (job.PipelineDepth > 0)
+            {
+                command += $" -- {job.PipelineDepth}";
+            }
+
+            var process = new Process()
+            {
+                StartInfo = {
+                    FileName = "stdbuf",
+                    Arguments = $"-oL {command}",
+                    WorkingDirectory = Path.GetDirectoryName(typeof(Startup).GetTypeInfo().Assembly.Location),
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true
+                },
+                EnableRaisingEvents = true
+            };
+
+            process.OutputDataReceived += (_, e) =>
+            {
+                Log(e.Data);
+                job.Output += (e.Data + Environment.NewLine);
+            };
+
+            process.ErrorDataReceived += (_, e) =>
+            {
+                Log(e.Data);
+                job.Error += (e.Data + Environment.NewLine);
+            };
+
+            process.Exited += (_, __) =>
+            {
+                double rps = -1;
+                var match = Regex.Match(job.Output, @"Requests/sec:\s*([\d.]*)");
```

Comment:
Allocations on this are going to be :banana:s - did you learn nothing from my talk? :trollface: